cmd/aterm/appdialogs/first_run_prompt.go

```diff
@@ -0,0 +1,188 @@
+
+	checkConnection := true
+
+	for checkConnection {
+		fmt.Printf(fancy.ClearLine("Let's check the network connection\n"))
+
+		network.SetBaseURL(*configData.APIURL)
+		network.SetAccessKey(*configData.AccessKey)
+		network.SetSecretKey(*configData.SecretKey)
+
+		var testErr error
+		dialog.DoBackgroundLoading(dialog.SyncedFunc(func() {
+			_, testErr = network.TestConnection()
+		}))
+		if testErr == nil {
+			fmt.Printf("These configurations work.\n")
+			checkConnection = false
+		} else if errors.Is(testErr, network.ErrorConnectionNotFound) {
+			fmt.Printf("It looks like the server is not up or the address is wrong.\n")
+			fix, err := dialog.YesNoPrompt("Do you want to try to fix this?", "", medium)
+			if fix && err == nil {
+				configData.APIURL = askForAPIURL(configData.APIURL)
+			} else {
+				checkConnection = false
+			}
+		} else if errors.Is(testErr, network.ErrorConnectionUnauthorized) {
+			fmt.Printf("The server did not accept your access and secret key.\n")
+			fix, err := dialog.YesNoPrompt("Do you want to try to fix this?", "", medium)
+			if fix && err == nil {
+				configData.AccessKey, configData.SecretKey = askForAccessKeyAndSecret(configData.AccessKey, configData.SecretKey)
+			} else {
+				checkConnection = false
+			}
+		} else {
+			fmt.Printf("I got an error I wasn't expecting. It's: '%v'. "+
+				"This may be due to a network issue with the ASHIRT servers or with your own connection. "+
+				"Please try contacting an administrator for help.\n", testErr.Error())
+			checkConnection = false
+		}
+	}
+
+	fmt.Printf("\nOkay, that should be enough data for now. "+
+		"I will create a configuration file here: %v. "+
+		"You can find additional configuration options there.\n\n", config.ATermConfigPath())
+
+	return configData, nil
+}
+
+func askForAPIURL(guessValue *string) *string {
+	fmt.Println("I need to know how to reach the ASHIRT servers. This typically looks like: " +
+		"http://ashirt.company.com/api (though may not in your case). If you do not know, please contact your administrator.")
+
+	apiURL, err := queryWithDefault("Enter the API URL", guessValue)
+
+	if err != nil {
+		fmt.Println(fancy.Caution("I had a problem collecting the API URL", err))
+		return strPtr("")
+	}
+
+	return &apiURL
+}
+
+func askForAccessKeyAndSecret(apiKeyGuess, secretKeyGuess *string) (*string, *string) {
+	fmt.Println("I need to know your credentials to talk to the ASHIRT servers. " +
+		"You can generate a new key from your account settings on the ASHIRT website.")
+
+	apiKeyAnswer, err1 := queryWithDefault("Enter the Access Key", apiKeyGuess)
+	secretKeyAnswer, err2 := queryWithDefault("Enter the Secret Key", secretKeyGuess)
+
+	if err := multierror.Append(err1, err2).ErrorOrNil(); err != nil {
+		fmt.Println(fancy.Caution("I had a problem collecting this info", err))
+		return strPtr(""), strPtr("")
+	}
+	return &apiKeyAnswer, &secretKeyAnswer
+}
+
+func askForSavePath(guessValue *string) *string {
```

Comment:
Can we default this to the xdg path for home data (eg. $HOME/.local/share/aterm/recordings)

Reply:
looks like both of the xdg libraries I've looked at default to `~/Library/Application Support` for mac. Should I instead opt to use `~/.local/share` as the base (per XDG on linux)?